README.md

```diff
@@ -1,8 +1,129 @@
-# caramel.cpp
+# Caramel: A fast read-only key-value store
 
-This repository provides a C++ implementation of the [CARAMEL data structure](https://arxiv.org/pdf/2305.16545.pdf), a space-efficient read-only retrieval table with applications to data management problems in recommender systems and computational biology. 
+This repository provides a C++ implementation of the [CARAMEL data structure](https://arxiv.org/pdf/2305.16545.pdf), a space-efficient read-only retrieval table with applications to data management problems in recommender systems, computational biology, and more.
 
-## Development Scripts
+## What is CARAMEL?
+
```

Comment:
We may want to iterate on this section a bit. We use CARAMEL to refer to both the software library and a specific data structure. I worry that might be a little confusing.